realm/realm-library/src/main/java/io/realm/internal/OsRealmConfig.java

```diff
@@ -0,0 +1,248 @@
+
+        public byte getNativeValue() {
+            return value;
+        }
+    }
+
+    /**
+     * Builder class for creating {@code OsRealmConfig}. The {@code OsRealmConfig} instance should only be created by
+     * {@link SharedRealm}.
+     */
+    public static class Builder {
+        private RealmConfiguration configuration;
+        private OsSchemaInfo schemaInfo = null;
+        private SharedRealm.MigrationCallback migrationCallback = null;
+        private SharedRealm.InitializationCallback initializationCallback = null;
+        private boolean autoUpdateNotification = false;
+
+        /**
+         * Initialize a {@link OsRealmConfig.Builder} with a given {@link RealmConfiguration}.
+         */
+        public Builder(RealmConfiguration configuration) {
+            this.configuration = configuration;
+        }
+
+        /**
+         * Sets the schema which Object Store initializes the Realm with.
+         *
+         * @param schemaInfo {@code null} to initialize the Realm in dynamic schema mode. Otherwise Object Store will
+         *                   initialize the Realm with given schema and handle migration with it.
+         * @return this {@link OsRealmConfig.Builder}.
+         */
+        public Builder schemaInfo(OsSchemaInfo schemaInfo) {
+            this.schemaInfo = schemaInfo;
+            return this;
+        }
+
+        /**
+         * Sets the callback when manual migration needed.
+         *
+         * @param migrationCallback callback to be set.
+         * @return this {@link OsRealmConfig.Builder}.
+         */
+        public Builder migrationCallback(SharedRealm.MigrationCallback migrationCallback) {
+            this.migrationCallback = migrationCallback;
+            return this;
+        }
+
+        /**
+         * Sets the callback which will be called when the Realm is created and the schema has just been initialized.
+         *
+         * @param initializationCallback the callback to be set.
+         * @return this {@link OsRealmConfig.Builder}.
+         */
+        public Builder initializationCallback(SharedRealm.InitializationCallback initializationCallback) {
+            this.initializationCallback = initializationCallback;
+            return this;
+        }
+
+        /**
+         * Set to {@code false} to disable the background worker thread for producing change notifications. Change
+         * notifications are enabled by default.
+         *
+         * @param autoUpdateNotification {@code false} to disable. {@code true} to enable it.
+         * @return this {@link OsRealmConfig.Builder}.
+         */
+        public Builder autoUpdateNotification(boolean autoUpdateNotification) {
+            this.autoUpdateNotification = autoUpdateNotification;
+            return this;
+        }
+
+        // Package private because of the OsRealmConfig needs to carry the NativeContext. This should only be called
+        // by the SharedRealm.
+        OsRealmConfig build() {
```

Comment:
Nitpick. It feels a bit weird that the rest of  the methods are public, but this is not. Given that this class is already in `io.realm.internal` keeping this public would be fine, no?